app.js

```diff
@@ -1,63 +1,71 @@
 const fs = require("fs-extra");
 const path = require("path");
-const { getOptions, runFtpFunction } = require("./helpers");
+const { mapConnectOptions, runCallbackWithFtpClient } = require("./helpers");
 
 async function upload(action, settings) {
-  const options = getOptions(action, settings);
+  const connectOptions = mapConnectOptions(action, settings);
   const { localPath, remotePath } = action.params;
 
-  const stats = fs.statSync(localPath);
-  const remoteDir = stats.isDirectory() ? remotePath : path.dirname(remotePath);
+  const pathStat = await fs.promises.stat(localPath);
+  const remoteDirectory = pathStat.isDirectory() ? remotePath : path.dirname(remotePath);
 
-  const ftpFunc = async (client) => {
-    await client.ensureDir(remoteDir);
-    return stats.isDirectory()
+  const ftpCallback = async (client) => {
```

Comment:
A function can be used as a callback, when it's passed as a param to another function, but at the place of declaration it's still a function.
What does this function do? Reflect it in its name... or you could pass that directly to the function as anonymous lambda.